src/tsickle.ts

```diff
@@ -759,43 +797,87 @@ class Annotator extends ClosureRewriter {
       //
       // Both of these forms create a local name "a", which after
       // TypeScript CommonJS compilation will become some renamed
-      // variable like "module_1.a".  But a user might still use plain
-      // "a" in some JSDoc comment, so gather up these local names for
-      // imports and make an alias for each for JSDoc purposes.
+      // variable like "module_1.default" or "module_1.a" (for default vs named bindings,
+      // respectively).
+      // tsickle references types in JSDoc. Because the module prefixes are not predictable, and
+      // because TypeScript might remove imports entirely if they are only for types, the code below
+      // inserts an artificial `const prefix = goog.require` call for the module, and then registers
```

Comment:
Yeah, I didn't either when I filed the PR, that's what I meant with descriptions etc being a mess.